lib/cppcheck.rb

```diff
@@ -182,21 +186,57 @@ def args_builder_common()
     @config[:undefines]&.each do |undefine|
       args << "-U#{undefine}"
     end
-    
-    args += @config[:arguments]
+
+    @config[:arguments]&.each do |argument|
+      args << "#{argument}"
+    end
+
+    return args
   end
   
-  def args_builder_text()
-    return [
-      "--output-file=#{@text_artifact_filepath}"
-    ]
+  def args_builder_text(args)
```

Comment:
Any reason why it is now needed to pass arguments to these functions?
I don't see it necessary and prefer to keep this functions returning an array of arguments on its own rather than modifying one that is given.

Reply:
I am newbie with Ruby, what I know is from reading ceedling source files.

So I don't know exactly how array works on Ruby, I was just the way I found to implement.